neural_lam/vis.py

```diff
@@ -9,6 +12,129 @@
 from .datastore.base import BaseRegularGridDatastore
 
 
+def plot_on_axis(
+    ax,
+    da,
+    datastore,
+    vmin=None,
+    vmax=None,
+    ax_title=None,
+    cmap="plasma",
+    boundary_alpha=None,
+    crop_to_interior=False,
+):
+    """Plot weather state on given axis using datastore metadata.
+
+    Parameters
+    ----------
+    ax : matplotlib.axes.Axes
+        The axis to plot on. Should have a cartopy projection.
+    da : xarray.DataArray or np.ndarray
+        The data to plot. Should have shape (N_grid,).
```

Comment:
Is it a good idea (from a maintainability standpoint) to allow this to be either `xarray.DataArray` or `np.ndarray`? I am leaning towards just enforcing this to be an `xarray.DataArray`. I think that would simplify some parts of this function, and I believe all calls to this anyhow use dataarrays now.